### PR TITLE
Add amount field to donation

### LINKED ISSUE #100: Add amount field to the Donation domain model
- add the `amount` field to the Donation table
- this field is `nullable=False`
- make a new migration

### PR DESCRIPTION
**Pull request by peterghrong**
# Summary

Added amount field to donation
Made column nullable=false
Finished migration and confirmed with Curtis
	
## Test Plan

Passed all unittests
1. Tested serialize method, passed
2. Tested when amount -> nullable=False is violated, passed. 

Screen Shot 2021-02-02 at 11 04 00 AM

## Related Issues

Which issues does this PR resolve/work on?
Closes #112

### CODE REVIEW

File: backend/tests/test_donation.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import unittest
+from app import create_app, db
+from app.models import Donation
+import datetime
+from sqlalchemy.exc import IntegrityError
+
+
+class ContactTestCase(unittest.TestCase):
```
Comment (CurtisMIT) [inline backend/tests/test_donation.py:8]: @peterghrong I think you forgot to change the name of the class here.

**Review by CurtisMIT — CHANGES_REQUESTED** [review]
@peterghrong Besides the minor typo, the migration has not been generated yet. A new version in the migrations folder should be there if a migration has been generated. Pm me if the commands are not working. Nice job!

**Comment by celinehuang** [comment]
@peterghrong take a look at this - https://flask-migrate.readthedocs.io/en/latest/ should help you with making a migration :)

**Comment by celinehuang** [comment]
In the future, you can avoid the list of commits by working on your new code with the updated main - either making a branch directly from the version currently on main or rebasing. 

to rebase, it's similar to merging but provides a cleaner commit history (here's an [article](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) if you want to read up about it)

```
$ git checkout main
$ git pull
apply migrations if there are any but you can also do this on your feature branch
$ git checkout featurebranch
$ git fetch
$ git rebase origin/main
$ git push --force
```

**Comment by peterghrong** [comment]
> In the future, you can avoid the list of commits by working on your new code with the updated main - either making a branch directly from the version currently on main or rebasing.
> 
> to rebase, it's similar to merging but provides a cleaner commit history (here's an [article](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) if you want to read up about it)
> 
> ```
> $ git checkout main
> $ git pull
> apply migrations if there are any but you can also do this on your feature branch
> $ git checkout featurebranch
> $ git fetch
> $ git rebase origin/main
> $ git push --force
> ```
> 
> having a cleaner commit history makes it easier for your reviewers to chance what each commit was doing (sometimes catching bugs etc)

Yes sorry will do in future PRs

**Comment by CurtisMIT** [comment]
@peterghrong Was wondering whether you could attach a screenshot of the updated table?

**Comment by peterghrong** [comment]
> @peterghrong Was wondering whether you could attach a screenshot of the updated table?

Screen Shot 2021-02-07 at 3 25 50 PM

Screenshot taken from pgAdmin with updated amount field in the donation model

**Review by CurtisMIT — DISMISSED** [review]
LGTM 💯

**Review by CurtisMIT — CHANGES_REQUESTED** [review]
@peterghrong Sorry can you show me the data type of the amount column? Sorry should have been more specific.

**Comment by peterghrong** [comment]
> @peterghrong Sorry can you show me the data type of the amount column? Sorry should have been more specific.

No worries ! I should have thought about that too
Screen Shot 2021-02-10 at 10 39 48 AM

**Review by CurtisMIT — APPROVED** [review]
LGTM!

**Pull request merged**